controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -156,6 +156,9 @@ public String getProcessorName() {
                                 StoreException.create(StoreException.Type.OPERATION_NOT_ALLOWED, "Postponed "
                                         + event + " so that waiting processor" + waitingRequestProcessor + " can work. "));
                     }
+                }).exceptionally(e -> {
+                    resultFuture.completeExceptionally(e);
```

Comment:
Isn't it fine to return the future as we have completed exceptionally?

Reply:
the future called `resultFuture` is returned to the caller already. we are just completing it here.